baiji/copy.py

```diff
@@ -265,7 +265,15 @@ def execute(self):
             else:
                 raise InvalidSchemeException("Copy for URI Scheme %s to %s is not implemented" % self.task)
         except KeyNotFound:
-            raise KeyNotFound("Error copying {} to {}: Source doesn't exist".format(self.src.uri, self.dst.uri))
+            if self.dst.is_s3:
+                try:
+                    _ = self.dst.bucket
+                except KeyNotFound:
+                    raise  KeyNotFound("Error copying {} to {}: Destination bucket doesn't exist".format(self.src.uri, self.dst.uri))
+            if not self.src.exists():
+                raise KeyNotFound("Error copying {} to {}: Source doesn't exist".format(self.src.uri, self.dst.uri))
+            else:
+                raise KeyNotFound("Error copying {} to {}: Destination doesn't exist".format(self.src.uri, self.dst.uri))
```

Comment:
When is this happening? If the bucket doesn't exist, it will fail before. If it does exists, it should copy it fine, so it should be local I guess. And if local, baiji creates the full tree of directories needed (if permissions allow it)